src/plugins/kernel_install/arch.sh

```diff
@@ -61,38 +114,57 @@ function install_kernel()
     return 22
   fi
 
+  if [[ "$local" == 'vm' ]]; then
+    # Check if vm is mounted and get its path
+    if [[ $(findmnt "${configurations[mount_point]}") ]]; then
+      path_prefix="${configurations[mount_point]}"
+      
+      # Copy config file
+#      cp -v .config $path_prefix/boot/config-$name
+    else
+      complain "Did you check if your VM is mounted?"
+      return 125 # ECANCELED
+    fi
+  fi
+
   # Copy kernel image
-  if [[ -f "/boot/vmlinuz-$name" ]]; then
-    cmd="$sudo_cmd cp /boot/vmlinuz-$name /boot/vmlinuz-$name.old"
+  if [[ -f "$path_prefix/boot/vmlinuz-$name" ]]; then
+    cmd="$sudo_cmd cp $path_prefix/boot/vmlinuz-$name $path_prefix/boot/vmlinuz-$name.old"
     cmd_manager "$flag" "$cmd"
   fi
 
   if [[ ! -z "$local" ]]; then
     [[ -z "$architecture" ]] && architecture="x86_64"
-    cmd="$sudo_cmd cp -v arch/$architecture/boot/bzImage /boot/vmlinuz-$name"
+    cmd="$sudo_cmd cp -v arch/$architecture/boot/bzImage $path_prefix/boot/vmlinuz-$name"
     cmd_manager "$flag" "$cmd"
   else
-    cmd="$sudo_cmd cp -v vmlinuz-$name /boot/vmlinuz-$name"
+    cmd="$sudo_cmd cp -v vmlinuz-$name $path_prefix/boot/vmlinuz-$name"
     cmd_manager "$flag" "$cmd"
   fi
 
   # Update mkinitcpio
   if [[ ! -z "$local" ]]; then
-    cmd="$sudo_cmd cp -v $LOCAL_KW_ETC /etc/mkinitcpio.d/$name.preset"
+    cmd="$sudo_cmd cp -v $LOCAL_KW_ETC $path_prefix/etc/mkinitcpio.d/$name.preset"
     cmd_manager "$flag" "$cmd"
-    cmd="$sudo_cmd sed -i -e \"s/NAME/$name/g\" \"/etc/mkinitcpio.d/$name.preset\""
+    cmd="$sudo_cmd sed -i -e \"s/NAME/$name/g\" \"$path_prefix/etc/mkinitcpio.d/$name.preset\""
     cmd_manager "$flag" "$cmd"
   else
-    cp -v "$name.preset" /etc/mkinitcpio.d/
+    cp -v "$name.preset" $path_prefix/etc/mkinitcpio.d/
   fi
 
-  cmd="$sudo_cmd mkinitcpio -p $name"
-  cmd_manager "$flag" "$cmd"
+  if [[ "$local" == 'local' ]]; then
+    # Update mkinitcpio
+    cmd="$sudo_cmd mkinitcpio -p $name"
+    cmd_manager "$flag" "$cmd"
+  fi
+  
+  # If VM is mounted, umount before update boot loader
+  [[ $(findmnt "${configurations[mount_point]}") ]] && vm_umount
```

Comment:
When I deployed to a remote machine, I got this error:

```
distro_deploy.sh: line 162: vm_umount: command not found
```

I think vm_umount get invoke here. If you want to reproduce this bug:
1. Enable remote deploy
2. Turn a Arch VM machine on
3. After you have a connection, deploy it using the remote option.

You should see the error at the end of the deploy.